pika/adapters/blocking_connection.py

```diff
@@ -344,6 +342,8 @@ def _handle_read(self):
         keeps track of how many frames have been written since the last read.
 
         """
+        if not hasattr(self, '_read_poller'):
+            raise exceptions.ConnectionClosed()
```

Comment:
@vitaly-krugl, not sure I agree on that one, it fixes the issue it sets out
to fix and I don't think it introduces any new ones. It wasn't aiming to
address any broader issues in BlockingConnection which are the subject of
your PRs.

On Mon, 13 Apr 2015 at 18:40 vitaly-krugl <notifications@github.com> wrote:

> In pika/adapters/blocking_connection.py
> <https://github.com/pika/pika/pull/546#discussion_r28260003>:
>
> > @@ -344,6 +342,8 @@ def _handle_read(self):
> >          keeps track of how many frames have been written since the last read.
> >
> >          """
> > +        if not hasattr(self, '_read_poller'):
> > +            raise exceptions.ConnectionClosed()
>
> @wjps <https://github.com/wjps>, this is not the right fix because the
> raised exception would not be consistent with the state of the connection -
> i.e., connection.is_closed would still evaluate to False after this. It's
> the same problem as with the AttributeError exception handler that this
> PR removed.
>
> The issue is reported in #412 <https://github.com/pika/pika/issues/412>
> and I believe that PRs #533 <https://github.com/pika/pika/pull/533> and
> #542 <https://github.com/pika/pika/pull/542> together address this issue,
> while at the same time maintaining consistent connection state. I didn't
> remove the AttributeError block in those PRs, though.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pika/pika/pull/546/files#r28260003>.
>